tests/test_edgeql_ddl.py

```diff
@@ -6132,3 +6136,98 @@ async def test_edgeql_ddl_describe_migration_json_03(self):
         )
 
         self.assertEqual(json.loads(result), expected_1)
+
+    async def test_edgeql_ddl_describe_migration_json_04(self):
+        self.maxDiff = None
+        await self.migrate('''
+            type Test;
+        ''')
+
+        await self.con.execute('''
+            START MIGRATION TO {
+                module test {
+                    type Test2;
+                    type Test3;
+                };
+            };
+        ''')
+
+        expected = {
+            'parent': 'm1xh653zionj2aehqbh7x6km5lo3b2mjaftxdkvqoh3wluc3iv6k2a',
+            'confirmed': [],
+            'complete': False,
+            'proposed': {
+                'statements': [{
+                    'text': 'ALTER TYPE test::Test RENAME TO test::Test2;',
+                }],
+                'confidence': 1.0,
+                'operation_id': 'ALTER TYPE test::Test',
+                'prompt': (
+                    "did you rename object type 'test::Test' to 'test::Test2'?"
+                ),
+            },
+        }
+
+        result = await self.con.query_one(
+            '''
+                DESCRIBE CURRENT MIGRATION AS JSON;
+            ''',
+        )
+
+        self.assertEqual(json.loads(result), expected)
+
+        await self.con.execute('''
+            ALTER CURRENT MIGRATION REJECT PROPOSED;
+        ''')
+
+        expected_1 = {
+            'parent': 'm1xh653zionj2aehqbh7x6km5lo3b2mjaftxdkvqoh3wluc3iv6k2a',
+            'confirmed': [],
+            'complete': False,
+            'proposed': {
+                'statements': [{
+                    'text': 'ALTER TYPE test::Test RENAME TO test::Test3;',
+                }],
+                'confidence': 1.0,
+                'operation_id': 'ALTER TYPE test::Test',
+                'prompt': (
+                    "did you rename object type 'test::Test' to 'test::Test3'?"
+                ),
+            },
+        }
+
+        result = await self.con.query_one(
+            '''
+                DESCRIBE CURRENT MIGRATION AS JSON;
+            ''',
+        )
+
+        self.assertEqual(json.loads(result), expected_1)
+
+        await self.con.execute('''
+            ALTER TYPE test::Test RENAME TO test::Test3;
+        ''')
+
+        expected_2 = {
+            'parent': 'm1xh653zionj2aehqbh7x6km5lo3b2mjaftxdkvqoh3wluc3iv6k2a',
+            'confirmed': ['ALTER TYPE test::Test RENAME TO test::Test3'],
+            'complete': False,
+            'proposed': {
+                'statements': [{
+                    'text': 'CREATE TYPE test::Test2;',
+                }],
+                'confidence': 1.0,
+                'operation_id': 'CREATE TYPE test::Test2',
+                'prompt': (
+                    "did you create object type 'test::Test2'?"
```

Comment:
I'm not sure I following the test. There was the 'test::Test' type. We then migrated to 'test::Test2' and 'test::Test3' types, and confirmed that we renamed 'Test' to 'Test3'.  Why do we ask about 'Test2' now?

Reply:
Because that's how `DESCRIBE CURRENT MIGRATION` works.  It's the decision of the frontend whether or not to show the actual prompt, or just accept the `CREATE` (this is driven by `confidence` and `safe`).

Reply:
What happens if I reject this suggestion?

Reply:
The suggested list will be empty and the "complete" key will also be `false`, which would indicate to the frontend that the server has exhausted all its suggestions and the user must complete the migration manually (or rollback to an earlier point).